apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/TransactionImpl.java

```diff
@@ -231,33 +259,33 @@ BuilderImpl index(short index) {
 
     }
 
-    private final short deadline;
+    private short deadline;
     private volatile byte[] senderPublicKey;
-    private final long recipientId;
-    private final long amountATM;
-    private final long feeATM;
-    private final byte[] referencedTransactionFullHash;
-    private final TransactionType type;
-    private final int ecBlockHeight;
-    private final long ecBlockId;
-    private final byte version;
-    private final int timestamp;
-    private final byte[] signature;
-    private final Attachment.AbstractAttachment attachment;
-    private final Appendix.Message message;
-    private final Appendix.EncryptedMessage encryptedMessage;
-    private final Appendix.EncryptToSelfMessage encryptToSelfMessage;
-    private final Appendix.PublicKeyAnnouncement publicKeyAnnouncement;
-    private final Appendix.Phasing phasing;
-    private final Appendix.PrunablePlainMessage prunablePlainMessage;
-    private final Appendix.PrunableEncryptedMessage prunableEncryptedMessage;
-
-    private final List<Appendix.AbstractAppendix> appendages;
-    private final int appendagesSize;
+    private long recipientId;
+    private long amountATM;
+    private long feeATM;
+    private byte[] referencedTransactionFullHash;
+    private TransactionType type;
+    private int ecBlockHeight;
+    private long ecBlockId;
+    private byte version;
+    private int timestamp;
+    private byte[] signature;
+    private Attachment.AbstractAttachment attachment;
```

Comment:
Why did you remove final modifiers? It allows transactionImpl to be thread-sage

Reply:
It deosn't want to compile otherwise. That was teh reason to remove **final** here.

Reply:
non-final fields can cause some runtime errors.  We need to return final modifiers in any case.

Reply:
Done by you